File: lab3/psfilter.py
#!/usr/bin/python

import sys
import re

def filter(x):
    line = sys.stdin.readlines()
    for out in line:
        out = out.rstrip("\n")
        regex='^'+x
        if re.search(regex, out):
            out = re.split(r"\s+", out)
            print out[1],
            for i in range(7,len(out)):
                print out[i],
                if i+1 == len(out):
                    print ""

if len(sys.argv) == 2:
    filter(sys.argv[1])

elif len(sys.argv) == 1:
    filter("morinor")

else:
     print "Error! Usage: psfilter.py [username]"

File: lab2/psfilter.pl
#!/usr/bin/perl
#Author: Ryosuke Morino
#Description: ps filter

use warnings;
use strict;

if (@ARGV == 1)
{
	while (my $input = <STDIN>)
	{
		if ($input =~ /^$ARGV[0]/)
		{
			my @words = split(/\s+/, $input);
			print "$words[1]";
            		my $num = @words-1;
          		for ( my $i=7; $i<=$num; $i++ )
			{
				print " $words[$i]";
				if ($i == $num)
				{
					print "\n";	
				}
			}
		}
	}
}
elsif (@ARGV == 0)
{
	while (my $input = <STDIN>)
        {
		if($input =~ /^morinor/ )
		{
                        my @words = split(/\s+/, $input);
                        print "$words[1]";
			my $num = @words-1;
                        for ( my $i=7; $i<=$num; $i++ )
                        {
                                print " $words[$i]";
                                if ($i == $num)
                                {
                                        print "\n";
                                }
                        }


		}
        }
}

else
{
	die "Error! Usage: psfilter.pl [username]\n";
}

